File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.py
class Solution:
    def maxScore(self, nums: List[int]) -> int:
        #simple greedy approach
        nums.sort(reverse = True)
        sumOfNumbers = 0
        
        for index , num in enumerate(nums):
            sumOfNumbers += num
            if sumOfNumbers <= 0:
                return index
            
        return len(nums)
        

File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxScore = function(nums) {
    nums.sort((a,b) => b-a)
    let sumOfValues = 0
    let index = 0
    let size = nums.length

    while (index < size) {
        sumOfValues += nums[index]
        if (sumOfValues <= 0)
            return index
        index++
    }

    return size
};